### PR TITLE
[Merged by Bors] - Add support for LDAP authentication

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

Doesn't make to much sense without authZ, but that is another story. Trying to split that to keep PRs small.

TODOs:

- [x] Release operator-rs
- [x] Watch AuthenticationClass

I tried using multiple LoginIdentityProvider as follows but NIFI failed to start up
```
        <provider>
            <identifier>login-identity-provider</identifier>
            <class>org.apache.nifi.authentication.single.user.SingleUserLoginIdentityProvider</class>
            <property name="Username">sbernauer</property>
            <property name="Password">sbernauer</property>
        </provider>
        <provider>
            <identifier>login-identity-provider</identifier>
            <class>org.apache.nifi.ldap.LdapProvider</class>
            <property name="Authentication Strategy">{authentication_strategy}</property>

            <property name="Manager DN">xxx_ldap_bind_username_xxx</property>
            <property name="Manager Password">xxx_ldap_bind_password_xxx</property>

            <property name="Referral Strategy">THROW</property>
            <property name="Connect Timeout">10 secs</property>
            <property name="Read Timeout">10 secs</property>

            <property name="Url">{protocol}://{hostname}:{port}</property>
            <property name="User Search Base">{search_base}</property>
            <property name="User Search Filter">{search_filter}</property>

            <property name="TLS - Client Auth">NONE</property>
            <property name="TLS - Keystore">/stackable/keystore/keystore.p12</property>
            <property name="TLS - Keystore Password">secret</property>
            <property name="TLS - Keystore Type">PKCS12</property>
            <property name="TLS - Truststore">/stackable/keystore/truststore.p12</property>
            <property name="TLS - Truststore Password">secret</property>
            <property name="TLS - Truststore Type">PKCS12</property>
            <property name="TLS - Protocol">TLSv1.2</property>
            <property name="TLS - Shutdown Gracefully">true</property>

            <property name="Identity Strategy">USE_DN</property>
            <property name="Authentication Expiration">7 days</property>
        </provider>
```
Startup error
```
│ nifi 2022-07-27 13:52:22,343 INFO [NiFi logging handler] org.apache.nifi.StdOut Caused by: org.apache.nifi.authentication.exception.ProviderCreationException: The Authentication Expiration must be specified.                          │
│ nifi 2022-07-27 13:52:22,343 INFO [NiFi logging handler] org.apache.nifi.StdOut     at org.apache.nifi.ldap.LdapProvider.onConfigured(LdapProvider.java:79)

```
Which doesn't make any sense

## Review Checklist

- [x] CRD change approved
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added (or not applicable)
- [ ] Documentation added (or not applicable)
- [ ] Changelog updated (or not applicable)
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
- [ ] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

**Review by maltesander — CHANGES_REQUESTED** [review]
LGTM overall. One bigger thing, the `create_reporting_task_job` fails when using ldap:
```
cat: /stackable/adminuser/username: No such file or directory                                                   
cat: /stackable/adminuser/password: No such file or directory                                                   
usage: create_nifi_reporting_task.py [-h] -n NIFI_API_URL -u USERNAME -p                                        
                                     PASSWORD -v NIFI_VERSION -c CERT -m                                        
                                     METRICS_PORT [-t TASK_NAME]                                                
create_nifi_reporting_task.py: error: argument -u/--username: expected one argument
```
Not sure how, but it needs the right credentials from LDAP. Any idea how we can do that?

File: rust/crd/Cargo.toml [hunk 1]
```diff
@@ -8,11 +8,12 @@ repository = "https://github.com/stackabletech/nifi-operator"
...
 
 [dependencies]
+indoc = "1.0.6"
 xml-rs = "0.8"
 serde = "1.0"
 serde_json = "1.0"
 snafu = "0.7"
 rand = "0.8"
-stackable-operator = { git = "https://github.com/stackabletech/operator-rs.git", tag = "0.21.1" }
+stackable-operator = { git = "https://github.com/stackabletech/operator-rs.git", branch = "main" }
```
[the remark this answers is not in the document]
Comment (sbernauer) [reply rust/crd/Cargo.toml:17]: Nope, sadly not. I need https://github.com/stackabletech/operator-rs/commit/8c573874474d7f4b21719955844eb460d1f82d42

File: rust/operator-binary/src/controller.rs [hunk 2]
```diff
@@ -239,14 +235,16 @@ pub async fn reconcile_nifi(nifi: Arc<NifiCluster>, ctx: Context<Ctx>) -> Result
             let rg_log_configmap = build_node_rolegroup_log_config_map(&nifi, &rolegroup)?;
 
             let rg_statefulset = build_node_rolegroup_statefulset(
+                client,
```
[the remark this answers is not in the document]
Comment (sbernauer) [reply rust/operator-binary/src/controller.rs]: good point, implemented

**Comment by sbernauer** [comment]
The reporting job should now be fixed.

**Review by maltesander — APPROVED** [review]
LGTM. Maybe talk to @siegfriedweber  how and when we can release a new operator-rs tag?

**Comment by sbernauer** [comment]
@maltesander could you please do one final review?
I released operator-rs and tried using multiple LoginIdentityProviders without success (put the error in the description)

**Review by maltesander — APPROVED** [review]
LGTM.

**Pull request closed without merging**